src/transformers/modeling_bart.py

```diff
@@ -1085,20 +1081,14 @@ def _force_token_ids_generation(self, scores, token_id) -> None:
 
     @staticmethod
     def _reorder_cache(past, beam_idx):
-        ((enc_out, enc_mask), decoder_past_key_values) = past
         reordered_past = []
-        for layer_past in decoder_past_key_values:
+        for layer_past in past:
             # get the correct batch idx from decoder layer's batch dim for cross and self-attn
             layer_past_new = {
                 attn_key: _reorder_buffer(attn_cache, beam_idx) for attn_key, attn_cache in layer_past.items()
             }
             reordered_past.append(layer_past_new)
-
-        new_enc_out = enc_out if enc_out is None else enc_out.index_select(0, beam_idx)
```

Comment:
don't we need to reorder encoder outputs to line up with `beam_idx` or is that done elsewhere?

Reply:
I don't think we have to reorder the encoder outputs at all because each "new" beam has the exact same "encoder_outputs".

For example imagine we do beam search with batch size 2 and num_beams=2 -> input_ids will be of shape [["Hello this is longer", "Hello <PAD> <PAD> <PAD>"]] (this is batch size 2). If we select `num_beams =2` what happens in generate is that we broadcast the input_ids of batch size 2 to batch size 4 (num_beams * batch_size) meaning that each beam for each batch [["Hello this is longer", "Hello this is longer", "Hello <PAD> <PAD> <PAD>", "Hello <PAD> <PAD> <PAD>"]] will have the same encoder outputs. Hope this makes sense @sshleifer .

Also I ran `RUN_SLOW=1 test_modeling_bart` -> so it should be good

Reply:
That sounds reasonable, but I'd be careful!